src/main/scala/zio/prelude/NonEmptySet.scala

```diff
@@ -0,0 +1,129 @@
+package zio.prelude
+
+import scala.language.implicitConversions
+
+sealed abstract case class NonEmptySet[A] private (elem: A, others: Set[A]) extends (A => Boolean) {
```

Comment:
Since we are so reliant on the underlying representation of `Set` here I wonder if it makes more sense to model this more like we did `NonEmptyChunk` where you just have `final class NonEmptySet[A] private (private val set: Set[A])` and you just enforce the invariant that all of the constructors only return underlying sets that are not empty. That way we don't have to keep adding this one additional element to the set each time we want to do something.